torchtext/experimental/datasets/text_classification.py

```diff
@@ -81,8 +88,13 @@ def _setup_datasets(dataset_name, root='.data', ngrams=1, vocab=None,
         logging.info('Creating {} data'.format(item))
         data_iter = _create_data_from_iterator(vocab, iters_group[item], removed_tokens)
         for cls, tokens in data_iter:
-            data[item]['data'].append((torch.tensor(cls),
-                                       torch.tensor([token_id for token_id in tokens])))
+            tok_list = [tok_id for tok_id in tokens]
+            try:
+                data[item]['data'].append((torch.tensor(cls),
```

Comment:
In which cases will this conversion not succeed? Using try except is a pretty big hammer. ValueError could be raised by many functions.

Reply:
This will fail when `tokens` is a list of string(s), e.g. the new `empty_tokenizer` is used for the case when you don't even want to tokenize your data.